### DISCUSSION TITLE
How I install luarock magick package with Lua 5.4?

### DISCUSSION THREAD

**Discussion by yriveiro** [3rd/image.nvim#136] [Q&A]
Hi, 

I followed the instructions to install this plugin through the [https://github.com/3rd/image.nvim?tab=readme-ov-file#requirements](requirements) but I was stuck on the second step.

```sh
luarocks install magick

Error: No results matching query were found for Lua 5.4.
To check if it is available for other Lua versions, use --check-lua-versions.
```

Is it possible to install `image.nvim` on Macos Sonoma 14.4.1?

**Comment by Kotatsu1** [comment]
Same issue @yriveiro Did you find the answer?

**Reply by yriveiro** [reply]
Nop, seems that the package is not available form my lua version 😕.

**Reply by Kotatsu1** [reply]
So you didn't find any other options to open images in nvim?

**Reply by yriveiro** [reply]
Sadly, no.

**Comment by vladimir-skvortsov** [comment]
Hey, try
```bash
luarocks --lua-version=5.1 install magick
```

**Reply by yriveiro** [reply]
@vladimir-skvortsov's suggestion seems to work

```sh
11:17:13 ❯ luarocks --lua-version=5.1 install magick                                                                                                                                     
Installing https://luarocks.org/magick-1.6.0-1.src.rock

magick 1.6.0-1 depends on lua 5.1 (5.1-1 provided by VM: success)
No existing manifest. Attempting to rebuild...
magick 1.6.0-1 is now installed in /Users/yriveiro/.luarocks (license: MIT)
```

**Reply by yriveiro** [reply]
For people who could be in the same situation as me.

Once my default lua installation is 5.4, and this recipe installs packages for version 5.1, it is required to add the 5.1 paths to the `LUA_PATH`, otherwise `image.nvim` doesn't find the package.

**Reply by Nicknamely** [reply]
Thank you.
 Btw how do I add 5.1 paths to the LUA_PATH ?

**Reply by 3rd** [reply]
Hey, I recommend just switching to the magick_cli processor instead of struggling with the rock.

**Reply by Nicknamely** [reply]
[This one](https://github.com/nodejs/node-gyp) ?
I guess i should remove the the magick packae i have installed...
Is there any other step required ?
